File: src/VitalSynth.Core/Vitals.cs
namespace VitalSynth.Core;

/// <summary>
/// A snapshot of vital signs at a point in time.
/// Produced by the session on every HL7 tick.
/// </summary>
public readonly record struct Vitals(
    BeatsPerMinute HeartRate,
    Percent        Spo2,
    Celsius        TemperatureC,
    Scenario       Scenario,
    DateTime       TimestampUtc
);

File: src/VitalSynth.Core.FS/Vitals.fs
namespace VitalSynth.Core.FS

open System

/// Snapshot of vital signs at a point in time.
///
/// F# records are immutable by default — there is no way to mutate a field
/// after creation. The equivalent C# `readonly record struct` requires the
/// `readonly` keyword and explicit init-only properties.
type Vitals = {
    HeartRate    : float<bpm>
    Spo2         : float<pct>
    TemperatureC : float<degC>
    Scenario     : Scenario
    TimestampUtc : DateTime
}
